vmm/src/signal_handler.rs

```diff
@@ -144,8 +170,8 @@ mod tests {
         assert!(filter.apply().is_ok());
         assert_eq!(METRICS.seccomp.num_faults.count(), 0);
 
-        // Calls the blacklisted SYS_getpid.
-        let _pid = process::id();
+        // Call the blacklisted `SYS_mkdir`.
```

Comment:
TLDR I allowed `getpid` and `kill` and changed the blackisted syscall from `getpid` to `mkdir` to make the GNU test pass consistently.

Long version:

* The glibc `SYS_getpid` caches its result, so it doesn't always do the syscall. Therefore, `process::id` didn't *always* break the seccomp filter on GNU - depending on whether `SYS_getpid` had been called before.
* I switched to:
  ```rust
  syscall(libc::SYS_getpid); // -> seccomp violation, handled
  ...
  raise(SIGBUS); // -> signal handler called
  ```
  That didn't work because glibc's `raise` calls *both* `getpid` and `gettid` (in some implementations - for instance, on the test machine it does, on mine it doesn't) so `getpid` had to be whitelisted. That's when I switched to whitelisting `getpid` and `gettid` for GNU - but:
  * the code became overly complicated with `#[cfg]`s for this simple test;
  * the test still failed on the test machine - despite `gettid` being whitelisted, it still caused a `SIGSYS` which, even more weirdly, did *not* get handled by the signal handler and terminated the test.
* At this point I decided to eliminate `#[cfg]`s and workarounds and:
  * cause a `SIGSYS` with `mkdir` (instead of `getpid`);
  * emit a `SIGBUS` with `kill` (instead of `raise`);
  *  whitelist `kill` and `getpid` (for the `kill` to go through).
  This solution works on both GNU and musl.

Reply:
Leaving this clarification here:
> whitelist kill and getpid (for the kill to go through).

is done just in a particular unit-test, not whitelisted in general.